web/src/fetch_status.ts

```diff
@@ -29,47 +37,47 @@ export class FetchStatus {
     // (described in detail below).
     _expected_max_message_id = 0;
 
-    start_older_batch(opts) {
+    start_older_batch(opts: FetchStatusOptions): void {
         this._loading_older = true;
         if (opts.update_loading_indicator) {
             message_feed_loading.show_loading_older();
         }
     }
 
-    finish_older_batch(opts) {
+    finish_older_batch(opts: FetchStatusOptions): void {
         this._loading_older = false;
-        this._found_oldest = opts.found_oldest;
-        this._history_limited = opts.history_limited;
+        this._found_oldest = opts.found_oldest ?? false;
+        this._history_limited = opts.history_limited ?? false;
```

Comment:
You probably can remove these given that I think these parameters will be non-optional if typed correctly.